src/__init__.py

```diff
@@ -7173,6 +7173,51 @@ def _validate(self):
 
         self._checker()  # any field_type specific checks
 
+    def _sync_flags(self):
+        """Propagate the field flags.
+
+        If this widget has a "/Parent", set its field flags and that of all
+        its /Kids widgets to the value of the current widget.
+        Only possible for widgets existing in the PDF.
+
+        Returns True or False.
+        """
+        if not self.xref:
+            return False  # no xref: widget not in the PDF
+        doc = self.parent.parent  # the owning document
+        if not doc:  # should never happen
+            return False
+        pdf = _as_pdf_document(doc)
+        # load underlying PDF object
+        pdf_widget = mupdf.pdf_load_object(pdf, self.xref)
+        Parent = mupdf.pdf_dict_get(pdf_widget, PDF_NAME("Parent"))
+        if not Parent.pdf_is_dict():
+            return False  # no /Parent: nothing to do
+
+        # put the field flags value into the parent field flags:
+        Parent.pdf_dict_put_int(PDF_NAME("Ff"), self.field_flags)
+
+        # also put that value into all kids of the Parent
+        kids = Parent.pdf_dict_get(PDF_NAME("Kids"))
+        if not kids.pdf_is_array():
+            return False  # no /Kids: should never happen!
```

Comment:
Could we use an assert here? Our return value is not checked by the caller, so this will silently fail.

Reply:
I am reluctant here:
After all, this may be an ill-formed PDF. We might want to issue a warning instead.
The other case (doc not available) would be a genuine PyMuPDF error.